cli/internal/runsummary/run_summary.go

```diff
@@ -105,6 +106,7 @@ func NewRunSummary(
 			EnvMode:           globalEnvMode,
 			Tasks:             []*TaskSummary{},
 			GlobalHashSummary: globalHashSummary,
+			GitState:          getGitState(repoRoot),
```

Comment:
one call out here is that this happens before runs. i chose to do it at the start, in case there are some git state modifying behaviors during the run (unlikely and likely unintentional), but we should try to capture up front)

Reply:
This doesn't really bother me a ton.